### PR TITLE
change field on create order

### PR DESCRIPTION
**Pull request by MaxAstin**

### CODE REVIEW

File: app/src/main/java/com/bunbeauty/papakarlo/feature/createorder/CreateOrderFragment.kt [hunk 1]
```diff
@@ -138,9 +145,16 @@ class CreateOrderFragment :
                         messageStringId = R.string.error_select_payment_method,
                         isShown = viewState.isPaymentMethodErrorShown
```
Comment (xidealo) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/createorder/CreateOrderFragment.kt:146]: 2 times we send viewState

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt [hunk 2]
```diff
@@ -127,12 +127,23 @@ class CreateOrderViewModel(
                 changePaymentMethod(paymentMethodUuid = action.paymentMethodUuid)
             }
 
+            is CreateOrder.Action.ChangeWithoutChangeChecked -> {
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:130]: Vary

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt [hunk 3]
```diff
@@ -334,13 +360,33 @@ class CreateOrderViewModel(
             return
         }
 
+        val newFinalCost = (state.cartTotal as? CreateOrder.CartTotal.Success)?.newFinalCostValue ?: 0
+        val isChangeIncorrect = state.paymentByCash &&
+            !state.withoutChangeChecked &&
+            ((state.change ?: 0) < newFinalCost)
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:366]: extract to varible pls

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt [hunk 4]
```diff
@@ -334,13 +360,33 @@ class CreateOrderViewModel(
...
+        }
+
         withLoading {
             if (userInteractor.isUserAuthorize()) {
                 val orderCode = createOrder(
                     isDelivery = state.isDelivery,
                     selectedUserAddress = state.selectedUserAddress,
                     selectedCafe = state.selectedCafe,
-                    orderComment = state.comment,
+                    orderComment = getExtendedComment(
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:383]: add trim

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt [hunk 5]
```diff
@@ -450,11 +496,35 @@ class CreateOrderViewModel(
...
+    private fun getExtendedComment(
+        state: CreateOrder.DataState,
+        withoutChange: String,
+        changeFrom: String,
+    ): String {
+        return buildString {
+            state.comment.takeIf { comment ->
+                comment.isNotBlank()
+            }?.let { comment ->
+                append("$comment ")
```
Comment (xidealo) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:514]: extract " " in paymentByCash
for example append(" ") or append(" (")

File: app/src/main/java/com/bunbeauty/papakarlo/feature/createorder/CreateOrderFragment.kt (hunk 1 above)
Comment (MaxAstin) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/createorder/CreateOrderFragment.kt]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt (hunk 3 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt (hunk 4 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:384]: fixed

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt (hunk 5 above)
Comment (MaxAstin) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/createorder/CreateOrderViewModel.kt:515]: add trim

**Review by xidealo — APPROVED** [review]

**Pull request merged**